File: solutions/javascript/recursion/mergeTwoSortedLists.js
const mergeTwoLists = (listOne, listTwo) => {
    if (listOne === null) return listTwo;
    if (listTwo === null) return listOne;

    if (listOne.val < listTwo.val) {
        listOne.next = mergeTwoLists(listOne.next, listTwo);
        return listOne;
    }

    listTwo.next = mergeTwoLists(listOne, listTwo.next);
    return listTwo;
};

File: solutions/python/recursion/merge_two_sorted_lists.py
class Solution:
    def mergeTwoLists(self, list_one, list_two):
        if not list_one:
            return list_two
        if not list_two:
            return list_one
            
        if list_one.val < list_two.val:
            list_one.next = self.mergeTwoLists(list_one.next, list_two)
            return list_one
        
        list_two.next = self.mergeTwoLists(list_one, list_two.next)
        return list_two